### DISCUSSION TITLE
Add unique element for array type

### DISCUSSION THREAD

**Discussion by intoxicated** [meilisearch/product#568] [Feedback & Feature Proposal]
I'm trying to migrate from Algolia to meilisearch, but I have been struggling to find a way to add unique element to array type.
For instance, let's say I have a document following:

```
{
   "id": 1234,
   "colors": ["red", "blue"]
}
```

and later I want to add some element to `colors` field but I wouldn't know which colors are already presented in the property unless I get the document first. 

Algolia provides a way to add unique element to array field with batch like
```
{
  "op": "update",
  "partialUpdate": "addUnique",
  "id": 1234,
  "field_name": "colors",
  "value": "red"
}
```
so this document will attempt to update existing document's `colors` field with `red`. Since, `red` is already present, it won't do anything. Does meilisearch provide somewhat similar feature? if not, what's best way to update array instead update whole property

**Comment by intoxicated** [comment]
or maybe like https://discuss.elastic.co/t/how-to-update-a-object-within-array-in-elastic-search/20115/8

**Comment by oluademola** [comment]
Hello @intoxicated 

Sorry, this is not possible at the moment. We appreciate your feedback, and we would include this in this existing [discussion](https://github.com/meilisearch/product/discussions/522) as part of our efforts to centralize feedback on this topic. 

Thank you very much.

**Reply by gmourier** [reply]
Thanks, @intoxicated and @oluademola

I'm locking that discussion since we moved the feedback [here](https://github.com/meilisearch/product/discussions/522)
